api/src/main/java/org/openmrs/api/db/hibernate/HibernatePatientDAO.java

```diff
@@ -425,85 +431,177 @@ public PatientIdentifierType getPatientIdentifierType(Integer patientIdentifierT
 	@SuppressWarnings("unchecked")
 	public List<Patient> getDuplicatePatientsByAttributes(List<String> attributes) {
 		List<Patient> patients = new Vector<Patient>();
-		
+		List<Integer> patientIds = new Vector<Integer>();
+
 		if (attributes.size() > 0) {
-			String select = "select distinct p1 from Patient p1, Patient p2";
-			String where = " where p1 <> p2 ";
-			String orderBy = " order by ";
-			
-			Class patient = Patient.class;
-			Set<String> patientFieldNames = new HashSet<String>(patient.getDeclaredFields().length);
-			for (Field f : patient.getDeclaredFields()) {
-				patientFieldNames.add(f.getName());
-				log.debug(f.getName());
-			}
-			
-			Class person = Person.class;
-			Set<String> personFieldNames = new HashSet<String>(person.getDeclaredFields().length);
-			for (Field f : person.getDeclaredFields()) {
-				personFieldNames.add(f.getName());
-				log.debug(f.getName());
+
+			String sqlString = getDuplicatePatientsSQLString(attributes);
+
+			try {
+				SQLQuery sqlquery = sessionFactory.getCurrentSession().createSQLQuery(sqlString);
+				patientIds = sqlquery.list();
+				if (patientIds.size() >= 1) {
+					Query query = sessionFactory.getCurrentSession().createQuery(
+							"from Patient p1 where p1.patientId in (:ids)");
+					query.setParameterList("ids", patientIds);
+					patients = query.list();
+				}
 			}
-			
-			Class personName = PersonName.class;
-			Set<String> personNameFieldNames = new HashSet<String>(personName.getDeclaredFields().length);
-			for (Field f : personName.getDeclaredFields()) {
-				personNameFieldNames.add(f.getName());
-				log.debug(f.getName());
+			catch (HibernateException ex) {
+				log.warn("Hibernate exception: " + ex.getStackTrace().toString());
+				throw ex;
+
 			}
-			
-			Class identifier = PatientIdentifier.class;
-			Set<String> identifierFieldNames = new HashSet<String>(identifier.getDeclaredFields().length);
-			for (Field f : identifier.getDeclaredFields()) {
-				identifierFieldNames.add(f.getName());
-				log.debug(f.getName());
+			finally {}
```

Comment:
:-) agreed, i had initially added some code and then removed it, now it is not of much use, I shall remove this block.